core/src/impl/Kokkos_ViewTracker.hpp

```diff
@@ -89,6 +91,17 @@ struct ViewTracker {
   KOKKOS_INLINE_FUNCTION
   explicit ViewTracker(const track_type& tt) noexcept
       : m_tracker(tt, view_traits::is_managed) {}
+
+  //! Move constructor.
+  KOKKOS_INLINE_FUNCTION
+  ViewTracker(ViewTracker&& other) : m_tracker(std::move(other.m_tracker)) {}
```

Comment:
Isn't that the default behavior?

Reply:
I also thought it would work using `ViewTracker(ViewTracker&&) = default;`.

However, doing so breaks the test I added in `TestViewMove.hpp`:
```bash
Note: Google Test filter = *view_move*
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from openmp
[ RUN      ] openmp.view_move
/workspaces/kokkos/core/unit_test/TestViewMove.hpp:50: Failure
Expected equality of these values:
  view.use_count()
    Which is: 2
  0
/workspaces/kokkos/core/unit_test/TestViewMove.hpp:53: Failure
Expected equality of these values:
  tester.view.use_count()
    Which is: 2
  1
[  FAILED  ] openmp.view_move (0 ms)
[----------] 1 test from openmp (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] openmp.view_move

 1 FAILED TEST
```

For the test's output, my conclusion is that instead of being moved from one view to another, the view tracker was copied.

Reading [the rules for Implicitly-declared move constructor](https://en.cppreference.com/w/cpp/language/move_constructor), it seems that a defaulted move constructor is not eligible due to `ViewTracker` having user-declared copy constructors.

Am I wrong? Am I missing something?

Reply:
https://godbolt.org/z/saKdjrW3z shows that a `default` move constructor is selected even when a user-defined copy constructor is provided and this `default` move constructor calls the move constructor for member variables. Only when removing the `default` move constructor, the copy constructor is selected instead.